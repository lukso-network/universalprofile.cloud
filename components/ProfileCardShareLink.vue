<script setup lang="ts">
import { useElementSize } from '@vueuse/core'

type Props = {
  resolvedName?: string
}

const props = defineProps<Props>()
const buttonRef = ref()
const buttonWidth = ref('auto')
const buttonFullWidth = ref('auto')
const hover = ref(false)
const isCopied = ref(false)

const handleCopy = () => {
  navigator.clipboard.writeText(link.value)
  isCopied.value = true

  setTimeout(() => {
    hover.value = false
    isCopied.value = false
  }, 500)
}

const link = computed(() => {
  return `https://m.profile.link/${props.resolvedName}`
})

onMounted(async () => {
  await nextTick()

  buttonWidth.value = '0px'
  const { width } = useElementSize(buttonRef.value)
  buttonFullWidth.value = `${width.value + 10}px`
})
</script>

<template>
  <div v-if="isCopied" class="absolute right-4 top-4">
    <lukso-button
      size="small"
      variant="secondary"
      class="group"
      custom-class="px-0 overflow-hidden"
      @click="handleCopy"
      ><lukso-icon name="link-1" size="small" class="mx-1.5"></lukso-icon>
      <div
        class="paragraph-inter-12-medium pl-1 pr-3 transition-all duration-500 ease-out"
      >
        {{ $formatMessage('share_link_copied') }}
      </div></lukso-button
    >
  </div>
  <div
    v-else
    class="absolute right-4 top-4"
    @mouseover="hover = true"
    @mouseleave="hover = false"
  >
    <lukso-button
      size="small"
      variant="secondary"
      class="group"
      custom-class="px-0 overflow-hidden"
      @click="handleCopy"
      ><lukso-icon name="link-1" size="small" class="mx-1.5"></lukso-icon>
      <div
        ref="buttonRef"
        class="paragraph-inter-12-medium transition-all duration-500 ease-out"
        :style="{
          width: hover ? buttonFullWidth : buttonWidth,
        }"
      >
        <div class="invisible flex text-nowrap text-left group-hover:visible">
          {{ $formatMessage('share_link_copy') }}
          <div class="ml-1 text-neutral-50">
            {{ removeSchemaFromUrl(link) }}
          </div>
        </div>
      </div></lukso-button
    >
  </div>
</template>
