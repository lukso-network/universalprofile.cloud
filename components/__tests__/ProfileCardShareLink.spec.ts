import { renderSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it } from 'vitest'

import ProfileCardShareLink from '@/components/ProfileCardShareLink.vue'

describe('ProfileCardShareLink', () => {
  const profile: Profile = {
    profileLink: {
      resolved:
        'https://universalswaps.io/social/0x1234567890abcdef1234567890abcdef12345678',
      link: 'https://universalswaps.io/social/{address}',
      address: '0x1234567890abcdef1234567890abcdef12345678',
      checksummed: '0x1234567890abcdef1234567890abcdef12345678',
      isResolved: true,
    },
  }

  it('should render', async () => {
    const component = await renderSuspended(ProfileCardShareLink, {
      props: { profile },
    })

    expect(component.html()).toMatchSnapshot()
  })
})
