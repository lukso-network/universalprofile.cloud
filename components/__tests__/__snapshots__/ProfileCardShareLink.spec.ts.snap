// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileCardShareLink > should render 1`] = `
"<div id="test-wrapper">
  <div>
    <lukso-button size="small" variant="secondary" custom-class="px-0 overflow-hidden size-[30px]">
      <lukso-icon size="small" name="transaction-send" class="mx-1.5"></lukso-icon>
      <div class="paragraph-inter-12-medium transition-width duration-500 ease-out" style="width: auto;">
        <div class="inline-flex items-center text-nowrap">
          <div class="m-1 mr-1.5 text-neutral-50">universalswaps.io/social/{address}</div>
        </div>
        <div class="inline-flex">
          <div class="mr-1.5"></div>
        </div>
      </div>
    </lukso-button>
  </div>
</div>"
`;
